### DISCUSSION TITLE
Calling reset() in a child component with a separate useForm() resets the parent component form.

### DISCUSSION THREAD

**Discussion by dulev** [react-hook-form/react-hook-form#3306] [Q&A]
**Describe the bug**
I have two components - `Parent` and `Child`. Both have an input field that is registered. Both have a call to `useForm()`. When I call `reset` from the child component the parent's field is also reset.

Is this the expected behavior or am I missing something?

**To Reproduce**
1. Open the codesandbox
2. Enter a value in both fields
3. Click reset.

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-reset-strange-behavior-xuymg

**Expected behavior**
I expect that calling `reset` would only reset the fields that are registered through the same `useForm()` and not any other field.

- OS: Manjaro Linux
- Browser: Chrome
- Version: 86.0.4240.75 (Official Build) (64-bit)

**Comment by bluebill1049** [comment]
This is expected if you want to reset partial:

`reset({ ...getValues(), xxx })`

**Comment by dulev** [comment]
Calling this in the child only resets the parent state (and vice versa - calling this in the parent resets only the child state). 

How do you reset the child state from the child and the parent state from the parent?

**Comment by bluebill1049** [comment]
`reset` is simple it reset the entire form, like what I suggested above, if you want to reset only child/parent or whatever section of your form, this logic will need to live in your app.
